From 589c2f53acd547cb02f712df326097b9a89cb6e2 Mon Sep 17 00:00:00 2001
From: Grzegorz Jaszczyk <jaz@semihalf.com>
Date: Fri, 16 Sep 2016 16:25:18 +0200
Subject: [PATCH 0509/1345] arm64: dts: marvell: add vbus regulator for USB
 found on Armada 8040 DB board

commit  a59d53f0a21650e8c11c22f1ebd62a1c9b6a834b from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

USB vbus can be configured through the io-expanders found on the board. The
expander0 is responsible for configuring USBs from CP0, while the expander1
is in charge of USBs vbus of CP1.

Change-Id: If6ebbcc5ebc9573f71be078f9554f9fdf0c650c7
Signed-off-by: Grzegorz Jaszczyk <jaz@semihalf.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/32936
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-8040-db.dtsi |   58 +++++++++++++++++++++++
 1 file changed, 58 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi b/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
index 4d8d24d..4f8b107 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
@@ -117,8 +117,47 @@
 					reg = <0x50>;
 					pagesize = <32>;
 				};
+
+			};
+
+			/* CON9 */
+			usb3@500000 {
+				usb-phy = <&cpm_usb3h0_phy>;
+			};
+
+			/* CON10 */
+			usb3@510000 {
+				usb-phy = <&cpm_usb3h1_phy>;
 			};
 		};
+
+		cpm_usb3h0_phy: cpm_usb3_phy0 {
+			compatible = "usb-nop-xceiv";
+			vcc-supply = <&cpm_reg_usb3h0_vbus>;
+		};
+
+		cpm_reg_usb3h0_vbus: cpm_usb3-vbus0 {
+			compatible = "regulator-fixed";
+			regulator-name = "cpm-usb3-vbus-h0";
+			regulator-min-microvolt = <5000000>;
+			regulator-max-microvolt = <5000000>;
+			enable-active-high;
+			gpio = <&expander0 0 GPIO_ACTIVE_HIGH>;
+		};
+
+		cpm_usb3h1_phy: cpm_usb3_phy1 {
+			compatible = "usb-nop-xceiv";
+			vcc-supply = <&cpm_reg_usb3h1_vbus>;
+		};
+
+		cpm_reg_usb3h1_vbus: cpm_usb3-vbus1 {
+			compatible = "regulator-fixed";
+			regulator-name = "cpm-usb3-vbus-h1";
+			regulator-min-microvolt = <5000000>;
+			regulator-max-microvolt = <5000000>;
+			enable-active-high;
+			gpio = <&expander0 1 GPIO_ACTIVE_HIGH>;
+		};
 	};
 
 	cpn-110-slave {
@@ -135,6 +174,25 @@
 					pagesize = <32>;
 				};
 			};
+
+			/* CON9 */
+			usb3@500000 {
+				usb-phy = <&cps_usb3h0_phy>;
+			};
+		};
+
+		cps_usb3h0_phy: cps_usb3_phy0 {
+			compatible = "usb-nop-xceiv";
+			vcc-supply = <&cps_reg_usb3h0_vbus>;
+		};
+
+		cps_reg_usb3h0_vbus: cps_usb3-vbus0 {
+			compatible = "regulator-fixed";
+			regulator-name = "cps-usb3-vbus-h0";
+			regulator-min-microvolt = <5000000>;
+			regulator-max-microvolt = <5000000>;
+			enable-active-high;
+			gpio = <&expander1 0 GPIO_ACTIVE_HIGH>;
 		};
 	};
 };
-- 
1.7.9.5

